packages/pymedphys_base/src/pymedphys_base/deliverydata/exportable.py

```diff
@@ -0,0 +1,226 @@
+# Copyright (C) 2019 Cancer Care Associates
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version (the "AGPL-3.0+").
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License and the additional terms for more
+# details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+# ADDITIONAL TERMS are also included as allowed by Section 7 of the GNU
+# Affero General Public License. These additional terms are Sections 1, 5,
+# 6, 7, 8, and 9 from the Apache License, Version 2.0 (the "Apache-2.0")
+# where all references to the definition "License" are instead defined to
+# mean the AGPL-3.0+.
+
+# You should have received a copy of the Apache-2.0 along with this
+# program. If not, see <http://www.apache.org/licenses/LICENSE-2.0>.
+
+
+from collections import namedtuple
+import functools
+from typing import Union, Tuple
+
+import numpy as np
```

Comment:
There might be value in refactoring numpy out of this package... Given it's going to be right at the bottom of the dependency tree... But then again, maybe there's no harm in always requiring numpy, especially now that numpy works just fine with pyodide... Not sure if there is an environment we want to target that doesn't have numpy...

Will we ever want the opportunity to have packages out of pymedphys be on hardware using something like https://micropython.org/? ... I think we would like to not be locked out of the option...

...okay, maybe it is best at some point to refactor numpy out of here.

Reply:
For our user base, I would say that `numpy` has a level of ubiquity on par with the standard library. 
But if you want to cater to microcontroller programming (which is probably strictly in scope, but by a very fine line), I won't object to it.

Reply:
> For our user base, I would say that `numpy` has a level of ubiquity on par with the standard library.
> But if you want to cater to microcontroller programming (which is probably strictly in scope, but by a very fine line), I won't object to it.

Let's not cater to it for now. On the day we need it (if we do), we can progressively refactor out the code that is usable into a separate package. Much the same as what we did for pyodide.

Reply:
:+1: